src/components/VmUserMessages/index.js

```diff
@@ -3,120 +3,103 @@ import PropTypes from 'prop-types'
 
 import { connect } from 'react-redux'
 
-import style from './style.css'
+import { Notification, NotificationDrawer, MenuItem, Icon, Button } from 'patternfly-react'
 
-import Time from '../Time'
+import style from './style.css'
 
-import { clearUserMessages } from '../../actions/vm'
-import { hrefWithoutHistory } from '../../helpers'
+import { clearUserMessages, displayUserMessages, dismissUserMessage } from '../../actions'
+import { hrefWithoutHistory, getFormatedDateTime } from '../../helpers'
 import { msg } from '../../intl'
 
-const UserMessage = ({ record, id }) => {
-  // TODO: render record.type
-  return (
-    <li className={`list-group-item ${style.crop}`} title={record.message} data-toggle='tooltip'>
-      <span>
-        <pre className={style['message-box']} id={id}>
-          <Time time={record.time} cssClass={style['usermsg-time']} />
-          {record.message}
-        </pre>
-      </span>
-    </li>
-  )
+const UserMessage = ({ record, id, onDismissMessage }) => {
+  const time = getFormatedDateTime(record.get('time'))
+  return (<Notification seen>
+    <NotificationDrawer.Dropdown id={id}>
+      <MenuItem onClick={onDismissMessage}>
+        { msg.clear() }
+      </MenuItem>
+    </NotificationDrawer.Dropdown>
+    <Icon className='pull-left' type='pf' name='warning-triangle-o' />
+    <Notification.Content>
+      <Notification.Message>
+        {record.get('message')}
+      </Notification.Message>
+      <Notification.Info leftText={time.date} rightText={time.time} />
+    </Notification.Content>
+  </Notification>)
 }
 UserMessage.propTypes = {
   record: PropTypes.object.isRequired,
   id: PropTypes.string,
-}
-
-const ContactAdminInfo = ({ userMessages, id }) => {
-  if (userMessages.get('records').size === 0) {
-    return null
-  }
-
-  return (
-    <div className={style['contact-admin']} id={id}>
-      Contact your administrator in case of issues
-    </div>
-  )
-}
-ContactAdminInfo.propTypes = {
-  userMessages: PropTypes.object.isRequired,
-  id: PropTypes.string,
-}
-
-function spaceToNonbreakableSpace (text) {
-  return text.split(' ').join('\u00A0')
+  onDismissMessage: PropTypes.func.isRequired,
 }
 
 class VmUserMessages extends React.Component {
   constructor (props) {
     super(props)
     this.state = {
       show: false,
+      expand: false,
     }
   }
 
   render () {
-    const { userMessages, onClearMessages } = this.props
+    const { userMessages, onClearMessages, onDisplayMessages, onDismissMessage } = this.props
 
     const idPrefix = `usermsgs`
-    const onToggle = () => {
-      this.setState({ show: !this.state.show })
+    const handleToggle = () => {
```

Comment:
If you move this to the class level (and rebind against this in the constructor), the arrow function will not need to be recreated at each render.